ProcessLib/BoundaryCondition/UniformDirichletBoundaryCondition.cpp

```diff
@@ -11,70 +11,57 @@
 
 #include <algorithm>
 #include <vector>
-
 #include <logog/include/logog.hpp>
 
+#include "MeshGeoToolsLib/MeshNodeSearcher.h"
+#include "BoundaryConditionConfig.h"
+
 namespace ProcessLib
 {
-UniformDirichletBoundaryCondition::UniformDirichletBoundaryCondition(
-    GeoLib::GeoObject const& geometry, BaseLib::ConfigTree const& config)
-    : _geometry(geometry)
+std::unique_ptr<UniformDirichletBoundaryCondition>
+createUniformDirichletBoundaryCondition(
+    BoundaryConditionConfig const& config,
+    MeshGeoToolsLib::MeshNodeSearcher& searcher,
+    NumLib::LocalToGlobalIndexMap const& dof_table,
+    int const variable_id)
 {
     DBUG("Constructing UniformDirichletBoundaryCondition from config.");
     //! \ogs_file_param{boundary_condition__type}
-    config.checkConfigParameter("type", "UniformDirichlet");
+    config.config.checkConfigParameter("type", "UniformDirichlet");
```

Comment:
ConfigTree is rather fragile (that's why there are so many test cases for it): All of its methods have to interact nicely. So when deriving from ConfigTree one probably would have to make most methods of ConfigTree `final` in order to not accidentally break things or accidentally forget some checks in the derived implementation.
Furthermore derivation is not necessarily good design, and in this special case BCConfig comprises a ConfigTree, a geometry and a component index, which are three entirely distinct things, so I'd favour composition anyway.